### ISSUE TITLE
Observing double entry for network status in pod annotations w/ Multus ver3.4.1

### ISSUE THREAD

**Issue by mukesh-dua** [intel/multus-cni#468]
**What happened**: Creating a pod using Multus CNI ver 3.4.1 results into double entry for network status in pod annotations
```
Name:         test-pod
Namespace:    default
Priority:     0
Node:         compute-1/xx.xx.xx.xx
Start Time:   Sat, 28 Mar 2020 19:09:54 +0000
Labels:       app=test
Annotations:  cni.projectcalico.org/podIP: 172.16.100.218/32
              k8s.v1.cni.cncf.io/network-status:
                [{
                    "name": "k8s-pod-network",
                    "ips": [
                        "172.16.100.218"
                    ],
                    "default": true,
                    "dns": {}
                }]
              k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "k8s-pod-network",
                    "ips": [
                        "172.16.100.218"
                    ],
                    "default": true,
                    "dns": {}
                }]
```
**What you expected to happen**:
A single entry for network status in pod annotations

**How to reproduce it (as minimally and precisely as possible)**:
```
ubuntu@compute-1:~/community/helm-charts$ #State of Kubernetes cluster before deployment of Multus pods
ubuntu@compute-1:~/community/helm-charts$ kubectl get po -A -owide
NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE     IP               NODE          NOMINATED NODE   READINESS GATES
default       busybox-sleep-less                         1/1     Running   1904       22d     172.16.100.67    compute-2   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-4wkz9   0/1     Evicted   0          46d     <none>           compute-1   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-g4b9w   1/1     Running   0          10d     172.16.100.101   compute-2   <none>           <none>
kube-system   calico-node-bwz54                          1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   calico-node-m22f5                          1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   calicoctl                                  1/1     Running   0          24d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-fw8s6                   1/1     Running   0          10d     172.16.100.73    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-z75q8                   1/1     Running   0          10d     172.16.100.99    compute-2   <none>           <none>
kube-system   etcd-compute-1                           1/1     Running   17         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-apiserver-compute-1                 1/1     Running   83         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-controller-manager-compute-1        1/1     Running   258        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-h5t87                           1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-jjqpn                           1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-scheduler-compute-1                 1/1     Running   238        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-8wmzf              1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-rnjf7              1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-kzk5f       1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-prpfj       1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
tenant1-ns    busybox-sleep                              1/1     Running   1          22d     172.16.100.68    compute-2   <none>           <none>
tenant1-ns    tenant1-nginx-698ffff655-xllvf             1/1     Running   0          22d     172.16.100.64    compute-2   <none>           <none>

ubuntu@compute-1:~/community/helm-charts$ #Install helm charts for Multus using community image 3.4.1
ubuntu@compute-1:~/community/helm-charts$ helm install full-coral ./multus
NAME: full-coral
LAST DEPLOYED: Sat Mar 28 19:09:12 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
======
1. The following components have been deployed as part of this helm chart:
Cluster Role: multus
Cluster Role Binding: multus
Config Map: full-coral-multus-0.1.0-config
Custom Resource Definition: network-attachment-definitions.k8s.cni.cncf.io
Daemon Set: full-coral-multus-ds
Service Account: multus

You can now deploy any other CNI and create its Network Attachment Defintion.
---------

2. To uninstall helm chart use the command:
helm delete full-coral

You may have to manually delete CRD -
kubectl delete crd network-attachment-definitions.k8s.cni.cncf.io
---------

ubuntu@compute-1:~/community/helm-charts$ #Observe the deployment of multus pods - "full-coral-multus-ds-xxxxx"
ubuntu@compute-1:~/community/helm-charts$ kubectl get po -A -owide
NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE     IP               NODE          NOMINATED NODE   READINESS GATES
default       busybox-sleep-less                         1/1     Running   1904       22d     172.16.100.67    compute-2   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-4wkz9   0/1     Evicted   0          46d     <none>           compute-1   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-g4b9w   1/1     Running   0          10d     172.16.100.101   compute-2   <none>           <none>
kube-system   calico-node-bwz54                          1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   calico-node-m22f5                          1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   calicoctl                                  1/1     Running   0          24d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-fw8s6                   1/1     Running   0          10d     172.16.100.73    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-z75q8                   1/1     Running   0          10d     172.16.100.99    compute-2   <none>           <none>
kube-system   etcd-compute-1                           1/1     Running   17         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   full-coral-multus-ds-fmkml                 1/1     Running   0          5s      yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   full-coral-multus-ds-kwnj9                 1/1     Running   0          5s      xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-apiserver-compute-1                 1/1     Running   83         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-controller-manager-compute-1        1/1     Running   258        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-h5t87                           1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-jjqpn                           1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-scheduler-compute-1                 1/1     Running   238        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-8wmzf              1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-rnjf7              1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-kzk5f       1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-prpfj       1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
tenant1-ns    busybox-sleep                              1/1     Running   1          22d     172.16.100.68    compute-2   <none>           <none>
tenant1-ns    tenant1-nginx-698ffff655-xllvf             1/1     Running   0          22d     172.16.100.64    compute-2   <none>           <none>

ubuntu@compute-1:~/community/helm-charts$ #Observe deployment of CRD - "network-attachment-definitions.k8s.cni.cncf.io"
ubuntu@compute-1:~/community/helm-charts$ kubectl get crd
NAME                                             CREATED AT
bgpconfigurations.crd.projectcalico.org          2020-02-10T21:32:29Z
bgppeers.crd.projectcalico.org                   2020-02-10T21:32:29Z
blockaffinities.crd.projectcalico.org            2020-02-10T21:32:29Z
cephblockpools.ceph.rook.io                      2020-02-11T20:39:58Z
cephclients.ceph.rook.io                         2020-02-11T20:39:58Z
cephclusters.ceph.rook.io                        2020-02-11T20:39:58Z
cephfilesystems.ceph.rook.io                     2020-02-11T20:39:58Z
cephnfses.ceph.rook.io                           2020-02-11T20:39:58Z
cephobjectstores.ceph.rook.io                    2020-02-11T20:39:58Z
cephobjectstoreusers.ceph.rook.io                2020-02-11T20:39:58Z
clusterinformations.crd.projectcalico.org        2020-02-10T21:32:29Z
felixconfigurations.crd.projectcalico.org        2020-02-10T21:32:29Z
globalnetworkpolicies.crd.projectcalico.org      2020-02-10T21:32:29Z
globalnetworksets.crd.projectcalico.org          2020-02-10T21:32:29Z
hostendpoints.crd.projectcalico.org              2020-02-10T21:32:29Z
ipamblocks.crd.projectcalico.org                 2020-02-10T21:32:29Z
ipamconfigs.crd.projectcalico.org                2020-02-10T21:32:29Z
ipamhandles.crd.projectcalico.org                2020-02-10T21:32:29Z
ippools.crd.projectcalico.org                    2020-02-10T21:32:29Z
network-attachment-definitions.k8s.cni.cncf.io   2020-03-28T19:09:12Z
networkpolicies.crd.projectcalico.org            2020-02-10T21:32:29Z
networksets.crd.projectcalico.org                2020-02-10T21:32:29Z
objectbucketclaims.objectbucket.io               2020-02-11T20:39:58Z
objectbuckets.objectbucket.io                    2020-02-11T20:39:58Z
volumes.rook.io                                  2020-02-11T20:39:58Z
volumesnapshotclasses.snapshot.storage.k8s.io    2020-02-10T22:36:08Z
volumesnapshotcontents.snapshot.storage.k8s.io   2020-02-10T22:36:08Z
volumesnapshots.snapshot.storage.k8s.io          2020-02-10T22:36:08Z

ubuntu@compute-1:~/community/helm-charts$ #Check the definition of test-pod 
ubuntu@compute-1:~/community/helm-charts$ cat ../../test_pod.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
  labels:
    app: test
spec:
  containers:
  - name: test-container
    image: busybox
    command: ['sh', '-c', 'echo Hello from test container! && sleep 3600']

ubuntu@compute-1:~/community/helm-charts$ #Create test-pod
ubuntu@compute-1:~/community/helm-charts$ kubectl create -f ../../test_pod.yaml 
pod/test-pod created

ubuntu@compute-1:~/community/helm-charts$ #Check the state of cluster post deployment of test-pod
ubuntu@compute-1:~/community/helm-charts$ kubectl get po -A -owide
NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE     IP               NODE          NOMINATED NODE   READINESS GATES
default       busybox-sleep-less                         1/1     Running   1904       22d     172.16.100.67    compute-2   <none>           <none>
default       test-pod                                   1/1     Running   0          4s      172.16.100.218   compute-1   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-4wkz9   0/1     Evicted   0          46d     <none>           compute-1   <none>           <none>
kube-system   calico-kube-controllers-6b9d4c8765-g4b9w   1/1     Running   0          10d     172.16.100.101   compute-2   <none>           <none>
kube-system   calico-node-bwz54                          1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   calico-node-m22f5                          1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   calicoctl                                  1/1     Running   0          24d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-fw8s6                   1/1     Running   0          10d     172.16.100.73    compute-2   <none>           <none>
kube-system   coredns-5644d7b6d9-z75q8                   1/1     Running   0          10d     172.16.100.99    compute-2   <none>           <none>
kube-system   etcd-compute-1                           1/1     Running   17         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   full-coral-multus-ds-fmkml                 1/1     Running   0          45s     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   full-coral-multus-ds-kwnj9                 1/1     Running   0          45s     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-apiserver-compute-1                 1/1     Running   83         46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-controller-manager-compute-1        1/1     Running   258        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-h5t87                           1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-proxy-jjqpn                           1/1     Running   8          46d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-scheduler-compute-1                 1/1     Running   238        46d     xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-8wmzf              1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
kube-system   kube-sriov-cni-ds-amd64-rnjf7              1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-kzk5f       1/1     Running   0          6d16h   xx.xx.xx.xx    compute-1   <none>           <none>
kube-system   kube-sriov-device-plugin-amd64-prpfj       1/1     Running   0          16d     yy.yy.yy.yy    compute-2   <none>           <none>
tenant1-ns    busybox-sleep                              1/1     Running   1          22d     172.16.100.68    compute-2   <none>           <none>
tenant1-ns    tenant1-nginx-698ffff655-xllvf             1/1     Running   0          22d     172.16.100.64    compute-2   <none>           <none>

ubuntu@compute-1:~/community/helm-charts$ #Describe the pod just deployed with 2 annotations
ubuntu@compute-1:~/community/helm-charts$ kubectl describe po test-pod
Name:         test-pod
Namespace:    default
Priority:     0
Node:         compute-1/xx.xx.xx.xx
Start Time:   Sat, 28 Mar 2020 19:09:54 +0000
Labels:       app=test
Annotations:  cni.projectcalico.org/podIP: 172.16.100.218/32
              k8s.v1.cni.cncf.io/network-status:
                [{
                    "name": "k8s-pod-network",
                    "ips": [
                        "172.16.100.218"
                    ],
                    "default": true,
                    "dns": {}
                }]
              k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "k8s-pod-network",
                    "ips": [
                        "172.16.100.218"
                    ],
                    "default": true,
                    "dns": {}
                }]
Status:       Running
IP:           172.16.100.218
IPs:
  IP:  172.16.100.218
Containers:
  test-container:
    Container ID:  docker://065a79831ec9db6287aeb33d23bafec7a8614612623cfa9eacfc96a9fe8f0486
    Image:         busybox
    Image ID:      docker-pullable://busybox@sha256:b26cd013274a657b86e706210ddd5cc1f82f50155791199d29b9e86e935ce135
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      echo Hello from test container! && sleep 3600
    State:          Running
      Started:      Sat, 28 Mar 2020 19:09:56 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-pkm2n (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  default-token-pkm2n:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-pkm2n
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type    Reason     Age   From                  Message
  ----    ------     ----  ----                  -------
  Normal  Scheduled  20s   default-scheduler     Successfully assigned default/test-pod to compute-1
  Normal  Pulling    19s   kubelet, compute-1  Pulling image "busybox"
  Normal  Pulled     18s   kubelet, compute-1  Successfully pulled image "busybox"
  Normal  Created    18s   kubelet, compute-1  Created container test-container
  Normal  Started    18s   kubelet, compute-1  Started container test-container
```
**Anything else we need to know?**:
This works well with v3.4

**Environment**:
- Multus version 
```
image:
  registry: docker.io
  repository: nfvpe/multus
  tag: v3.4.1
```
- Kubernetes version (use `kubectl version`):
```
ubuntu@compute-1:~/community/helm-charts$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:23:21Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Primary CNI for Kubernetes cluster: Calico v3.9.5
- OS (e.g. from /etc/os-release):
```
ubuntu@lenlmsdev30:~/community/helm-charts$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

ubuntu@compute-1:~/community/helm-charts$ uname -a
Linux compute-1 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
- File of '/etc/cni/net.d/'
```
ubuntu@compute-1:~/community/helm-charts$ cat /etc/cni/net.d/00-multus.conf 
{
  "binDir":"/opt/cni/bin",
  "capabilities":
    {
      "portMappings":true
    },
  "clusterNetwork":"k8s-pod-network",
  "cniDir":"/var/lib/cni/multus",
  "cniVersion":"0.3.1",
  "confDir":"/etc/cni/net.d",
  "defaultNetwork":[],
  "delegates":[],
  "kubeconfig":"/etc/cni/net.d/multus.d/multus.kubeconfig",
  "logFile":"/var/log/multus.log",
  "logLevel":"panic",
  "name":"multus-cni-network",
  "namespaceIsolation":false,
  "readinessindicatorfile":"",
  "systemNamespaces":["kube-system"],
  "type":"multus"
}
```
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`): test-pod(details above)

- Helm charts used to deploy Multus CNI can be found at path:
https://github.com/k8snetworkplumbingwg/helm-charts/tree/master/multus

**Comment by dougbtv** [comment]
We discussed this one on the maintainers meeting, but, quick recap:

We had a mistake where we weren't aligned with the NPWG specification, especially related to having the annotation named `networks-status` where the spec stated `network-status` (our mistake being pluralized, where the proper one in the spec is not plural) -- so instead of just changing it, we added an additional annotation `network-status`, and we'll remove the deprecated `networks-status` in the future.

So, plan on using `network-status` as we'll wind up dropping `networks-status`.

**Comment by chestack** [comment]
NetworkStatusAnnot has been changed in https://github.com/k8snetworkplumbingwg/multus-cni/pull/434 which was merged  and cherry-picked into v3.4.1 on Mar 13th 2020
